crates/sui-framework/packages/dwallet-system/sources/sui_state_proof.move

```diff
@@ -0,0 +1,145 @@
+#[allow(unused_field)]
+module dwallet_system::sui_state_proof {
+
+    use dwallet::object::{Self, ID, UID};
+    use dwallet_system::dwallet::{Self, DWalletCap, MessageApproval};
+    use dwallet::tx_context::{TxContext};
+    use dwallet::transfer;
+    use dwallet::bcs;
+    use dwallet::event;
+
+
+
+    struct StateProofRegistry has key, store {
+            id: UID,
+            highest_epoch: u64,
+    }
+
+    struct StateProofConfig has key, store{
+        id: UID,
+        registry_id: ID,
+        package_id: vector<u8>,
+        event_type_layout: vector<u8>,
+        message_field_name: vector<u8>,
+    }
+
+    struct EpochCommittee has key, store {
+        id: UID,
+        committee: vector<u8>,
+    }
+
+
+    struct EpochCommitteeSubmitted has copy, drop {
+        epoch: u64,
+        epoch_committee_id: ID,
+    }
+
+    struct CapWrapper has key, store {
+        id: UID,
+        cap_id_sui: ID,
+        cap: DWalletCap,
+    }
+
+
+    // TODO add wittness here so it can be initialized only once??
+    public fun init_module(init_committee: vector<u8>, package_id: vector<u8>, event_type_layout: vector<u8>, message_field_name: vector<u8>, ctx: &mut TxContext) {
+        let registry = StateProofRegistry {
+            id: object::new(ctx),
+            highest_epoch: 0,
+        };
+
+        let config = StateProofConfig {
+            id: object::new(ctx),
+            registry_id: object::id(&registry),
+            package_id: package_id,
+            event_type_layout: event_type_layout,
+            message_field_name: message_field_name,
+        };
+
+        let first_committee = EpochCommittee {
+            id: object::new(ctx),
+            committee: init_committee,
+        };
+
+        event::emit(EpochCommitteeSubmitted {
+            epoch: 0,
+            epoch_committee_id: object::id(&first_committee),
+        });
+        
+        transfer::share_object(registry);
+        transfer::freeze_object(config);
+        transfer::freeze_object(first_committee);
+    }
+
+    
+
+    native fun sui_state_proof_verify_committee(prev_committee: vector<u8>, checkpoint_summary: vector<u8>): vector<u8>;
```

Comment:
Place all native imports in a single place (either in the top or bottom of the file)